### DISCUSSION TITLE
Android file access

### DISCUSSION THREAD

**Discussion by t-arn** [beeware/toga#1450] [Ideas]
How do we address the problem that on Android 11 or higher we cannot access files outside the app's own file space anymore?

https://developer.android.com/about/versions/11/privacy/storage

For example: How do I read a file from the "Download" folder or write a file to the "Download" folder?

One possible idea would be this:
https://github.com/beeware/toga/issues/1171

Are there other ideas how to tackle this problem?

**Comment by freakboy3742** [comment]
Short answer - I don't know, but we clearly need to do something.

The good news I can see in that document - `fopen` now has first class support, so the full form of #1171 (doing File IO through a proxy layer) may not be required.

However - this is an area where we need to take a deliberate and planned approach, with a view to the capabilities on every platform. I'm open to any suggestions and design proposals in this area.
